aspnetcore/performance/response-compression.md

```diff
@@ -0,0 +1,167 @@
+
+When a client can process compressed content, the client must inform the server of its capabilities by sending the `Accept-Encoding` header with the request. When a server sends compressed content, it must include information in the `Content-Encoding` header on how the compressed response is encoded. Content encoding designations are shown below indicating which ones are supported by the middleware.
+
+Content Encoding | Supported | Description
+:---: | :---: | ---
+`br` |  No | Brotli Compressed Data Format
+`compress` | No | UNIX "compress" data format
+custom | Yes | Developer provides the compression implementation. The client must be able to decompress the payload.
+`deflate` |  No | "deflate" compressed data inside the "zlib" data format 
+`exi` | No | W3C Efficient XML Interchange
+`gzip` | Yes (default) | GZip file format
+`identity` | Yes | "No encoding" identifier: The response must not be encoded.
+`pack200-gzip` | No | Network Transfer Format for Java Archives
+`*` | Yes | Any available content encoding not explicitly requested
+
+For more information, see the [IANA Official Content Coding List](http://www.iana.org/assignments/http-parameters/http-parameters.xml#http-content-coding-registry).
+
+The middleware is capable of reacting to quality value (qvalue, `q`) weighting when sent by the client. For more information, see [RFC 7231: Accept-Encoding](https://tools.ietf.org/html/rfc7231#section-5.3.4).
+
+Most compression algorithms are subject to a tradeoff between compression speed and the effectiveness of the compression. *Effectiveness* in this context refers to the size of the output after compression. The smallest size is achieved by the most *optimal* compression. The middleware defaults to the fastest compression level, which might not produce the most efficient compression. If the most efficient compression is desired, the middleware can be configured for optimal compression.
+
+The headers involved in requesting, sending, caching, and receiving compressed content are described below.
+
+Header | Role
+--- | ---
+`Accept-Encoding` | Sent by the client to the server to indicate which types of content encoding are acceptable.
+`Content-Encoding` | Sent from the server to the client to indicate the encoding of the content in the payload.
+`Content-Length` | When compression occurs, the the `Content-Length` header is removed, since the body content changes when the response is compressed.
+`Content-MD5` | When compression occurs, the `Content-MD5` header is removed, since the body content has changed and the hash is no longer valid.
+`Content-Type` | Specifies the MIME type of the content. MIME types are specified on the server for encodable content, so this information is used by the server to determine if a response can be compressed. The middleware includes a set of [default MIME types](#mime-types) that it will encode, but you can replace or add MIME types.
```

Comment:
Each response should have a content-type. The middleware will check this value to determine if the response should be compressed.